apps/src/applab/dataBrowser/AddTableRow.jsx

```diff
@@ -1,10 +1,32 @@
+import FirebaseStorage from '../firebaseStorage';
 import Radium from 'radium';
 import React from 'react';
+import { castValue, displayValue } from './dataUtils';
 import * as dataStyles from './dataStyles';
 
 const AddTableRow = React.createClass({
   propTypes: {
-    columnNames: React.PropTypes.array.isRequired
+    columnNames: React.PropTypes.array.isRequired,
+    tableName: React.PropTypes.string.isRequired
+  },
+
+  getInitialState() {
+    return { newRecord: {} };
```

Comment:
I think you could get rid of the newRecord, and just have everything live directly on state? Then in your handleChange you could do:
```
this.setState({
  [columnName]: castValue(event.target.value)
});